contracts/interchain-token-service/src/contract/execute.rs

```diff
@@ -158,7 +158,8 @@ fn gateway_token_transfer(
     match (gateway_denom, message) {
         (Some(denom), Message::InterchainTransfer { amount, .. }) => Ok(Some(Coin {
             denom: denom.to_string(),
-            amount: Uint128::try_from(*amount).change_context(Error::TransferAmountOverflow)?,
+            amount: Uint128::try_from(Uint256::from(*amount))
```

Comment:
what's the purpose of this change?

Reply:
nonempty::Uint256 -> Uint256 -> Uint128. I'll try to add a direct conversion method instead

Reply:
removing gateway token logic so this won't be needed anyways